packages/mui-base/src/SelectUnstyled/SelectUnstyled.types.ts

```diff
@@ -81,10 +86,26 @@ export interface SelectUnstyledOwnProps<TValue extends {}> extends SelectUnstyle
    * The default selected value. Use when the component is not controlled.
    */
   defaultValue?: TValue | null;
+  /**
+   * A function to convert the currently selected value to a string.
+   * Used to set a value of a hidden input associated with the select,
+   * so that the selected value can be posted with a form.
+   */
+  getSerializedValue?: (
+    option: SelectOption<TValue> | null,
+  ) => React.InputHTMLAttributes<HTMLInputElement>['value'];
   /**
    * Callback fired when an option is selected.
    */
   onChange?: (value: TValue | null) => void;
+  /**
+   * A function used to convert the option label to a string.
+   * It's useful when labels are elements and need to be converted to plain text
+   * to enable navigation using character keys on a keyboard.
```

Comment:
```suggestion
   * A function used to convert the option label to a string.
   * It's used when labels are elements and need to be converted to plain text
   * to enable filtering the option by typing in the text field.
```

Maybe? I would ask @samuelsycamore :)

Reply:
It's not really filtering, and there's no text field in a select. It's for moving highlight to the element that starts with a character pressed on a keyboard.

Reply:
Alright, fair enough. If we receive complain we can update later :)